tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,375 @@
+    # https://docs.pytest.org/en/latest/reference/reference.html#stash
+    request.node.stash[synced_repos] = {}
+    yield request.node.stash
+
+
+RHEL7_VER = '7.9'
+RHEL8_VER = '8.8'
+RHEL9_VER = '9.2'
+
+RHEL_REPOS = {
+    'rhel7_server': {
+        'id': 'rhel-7-server-rpms',
+        'name': f'Red Hat Enterprise Linux 7 Server RPMs x86_64 {RHEL7_VER}',
+        'releasever': RHEL7_VER,
+        'reposet': 'Red Hat Enterprise Linux 7 Server (RPMs)',
+        'product': 'Red Hat Enterprise Linux Server',
+    },
+    'rhel7_server_extras': {
+        'id': 'rhel-7-server-extras-rpms',
+        'name': 'Red Hat Enterprise Linux 7 Server - Extras RPMs x86_64',
+        'releasever': '7',
+        'reposet': 'Red Hat Enterprise Linux 7 Server - Extras (RPMs)',
+        'product': 'Red Hat Enterprise Linux Server',
+    },
+    'rhel8_bos': {
+        'id': 'rhel-8-for-x86_64-baseos-rpms',
+        'name': f'Red Hat Enterprise Linux 8 for x86_64 - BaseOS RPMs {RHEL8_VER}',
+        'releasever': RHEL8_VER,
+        'reposet': 'Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)',
+    },
+    'rhel8_aps': {
+        'id': 'rhel-8-for-x86_64-appstream-rpms',
+        'name': f'Red Hat Enterprise Linux 8 for x86_64 - AppStream RPMs {RHEL8_VER}',
+        'releasever': RHEL8_VER,
+        'reposet': 'Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)',
+    },
+    'rhel9_bos': {
+        'id': 'rhel-9-for-x86_64-baseos-rpms',
+        'name': f'Red Hat Enterprise Linux 9 for x86_64 - BaseOS RPMs {RHEL9_VER}',
+        'releasever': RHEL9_VER,
+        'reposet': 'Red Hat Enterprise Linux 9 for x86_64 - BaseOS (RPMs)',
+    },
+    'rhel9_aps': {
+        'id': 'rhel-9-for-x86_64-appstream-rpms',
+        'name': f'Red Hat Enterprise Linux 9 for x86_64 - AppStream RPMs {RHEL9_VER}',
+        'releasever': RHEL9_VER,
+        'reposet': 'Red Hat Enterprise Linux 9 for x86_64 - AppStream (RPMs)',
+    },
+}
+
+
+@pytest.fixture
+def function_leapp_cv(module_target_sat, module_sca_manifest_org):
+    logger.info('Creating Leapp Content View')
+    return module_target_sat.api.ContentView(organization=module_sca_manifest_org).create()
+
+
+@pytest.fixture(scope='module')
+def module_leapp_lce(module_target_sat, module_sca_manifest_org):
+    logger.info('Creating Leapp Lifecycle Environment')
+    return module_target_sat.api.LifecycleEnvironment(organization=module_sca_manifest_org).create()
+
+
+@pytest.fixture
+def function_leapp_ak(
+    module_target_sat, function_leapp_cv, module_leapp_lce, module_sca_manifest_org, upgrade_path
+):
+    logger.info(
+        'Creating Leapp Activation Key for RHEL %s -> %s',
+        upgrade_path['source_version'],
+        upgrade_path['target_version'],
+    )
+    # need to publish and promote cv on lce to get content available on client host
+    function_leapp_cv.publish()
+    cvv = function_leapp_cv.read().version[0]
+    cvv.promote(data={'environment_ids': module_leapp_lce.id, 'force': True})
```

Comment:
Can we move this CV specific steps into `function_leapp_cv` fixture?

Reply:
I think we had discussion around this change back in some comment, not remembered exactly.
But if we move above 3 lines inside `function_leapp_cv`, cv will create > publish & promote to module_lce without any content, Repos are yet to be enable and sync

Reply:
@Gauravtalreja1 I would not block this PR on it. Your suggestion would bring us more granularity into the fixture structure. Also, I do not see the eventual `function_leapp_cv` fixture being used anywhere else than in `leapp_sat_content` fixture.

Reply:
Sure I will test and do necessary changes

Reply:
Thanks @vijaysawant :)